src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -1904,6 +1905,16 @@ public void stopBuild(@NonNull UUID studyUuid, @NonNull UUID nodeUuid) {
         networkModificationService.stopBuild(studyUuid, nodeUuid);
     }
 
+    public UUID duplicateStudyNode(UUID studyUuid, UUID nodeToCopyUuid, UUID referenceNodeUuid, InsertMode insertMode) {
+        checkStudyContainsNode(studyUuid, nodeToCopyUuid);
```

Comment:
what entities are fetched more than once ?
Also what is the problem with the annotation postCompletion

Reply:
He is saying that calling notificationService.emitNodeInserted inside the transaction is OK because the annotation makes it execute after the transaction anyway